wagtail/embeds/blocks.py

```diff
@@ -12,9 +15,10 @@ class EmbedValue:
     """
     def __init__(self, url):
         self.url = url
+        self.html = embed_to_frontend_html(url)
```

Comment:
I'm not keen on fetching the HTML on the constructor here, could we instead create ``html`` as a cached property?

Reply:
No problem with making it a cached property, but why would it be problematic to do it on initialisation?

Reply:
I think the lookup should be done during the call to ``{% embed %}`` and not at the point the ``EmbedValue`` is constructed. That's where I think most developers would expect it to happen.

Also, if the embed block is called like this: ``{% embed value.url 500 %}``, it would end up fetching the embed twice.

Reply:
Well, the whole point of this PR is to fetch it on save, so you can actually tell users if the URL is valid or not. Which is exactly what happens in rich text 😉

Reply:
That’s indeed what I wanted to do at first.
But if we do that, we end up with an `Embed` object and we have to catch `EmbedException`.
And when we want to display it in front-end, we have to use `render_to_string(…)` on that `Embed` object, or reuse `embed_to_frontend_html`.

So… The simplest solution was to directly reuse `embed_to_frontend_html`.  Or it would require to modify the whole `wagtail.embeds.format` file, which was out of scope and not required for a clean solution.

Reply:
> But if we do that, we end up with an Embed object and we have to catch EmbedException

Yep, what's the problem with that?

Reply:
The problem is simple, if we call `get_embed` in `EmbedValue.__init__` and `embed_to_frontend_html` in `EmbedValue.__str__`, then we end up with two queries for each embed displayed in the front-end.

Reply:
The issue I see with populating `self.html` on init is that we're going from a 'lazy evaluation' model to an eager one, with a consequent (potential) performance hit. Suppose someone had a 'random video' block consisting of a list of 100 Youtube URLs, from which it will pick one to display; calling `embed_to_frontend_html` on init will result in 100 queries rather than just the one.

Reply:
Then the solution is simply to move `html` to a cached property, isn’t it?